BuildHelpers/Public/Get-BuildEnvironment.ps1

```diff
@@ -107,6 +108,11 @@ function Get-BuildEnvironment {
         PSModuleManifest = ${Build.ManifestPath}
         ModulePath = ${Build.ModulePath}
     }
+    if (${Build.Vars}.IsPullRequest) {
+        #This will ensure IsPullRequest is inserted into the ordered table after the BuildNumber property
+        $index = $($BuildHelpersVariables.keys).IndexOf("BuildNumber")
```

Comment:
It's late so may be missing it, but any reason not to add IsPullRequest to the ordered hash definition above rather than this extra logic?

Reply:
My intention here (and answer to the rest of your questions sort of flow from this) is that since an Environment Variable can't be a boolean, just some text, that `IsPullRequest` would only exist if the run was a pull request, and so you could check just for existence of the environment variable.  Here, I wanted to ensure that it would appear after `BuildNumber` in the ordered hash, but I couldn't just add it to the hash definition because it wouldn't always be defined.  This addition here will add it into the ordered hash after BuildNumber, but only do so if there actually is a value